test/extensions/filters/http/aws_lambda/test_transformer.h

```diff
@@ -13,12 +13,23 @@ class FakeTransformer : public HttpFilters::Transformation::Transformer {
 public:
   bool passthrough_body() const override {return false;}
   // This transformer just drains the body and replaces it with a hardcoded string.
-  void transform (Http::RequestOrResponseHeaderMap &,
+  void transform (Http::RequestOrResponseHeaderMap &headers,
                          Http::RequestHeaderMap *,
                          Buffer::Instance &body,
                          Http::StreamFilterCallbacks &) const override {
+                            std::string *headers_string = new std::string("headers:\n");
+                            headers.iterate(
+                                [headers_string](const Http::HeaderEntry &header) -> Http::HeaderMap::Iterate {
+                                    auto key = std::string(header.key().getStringView());
+                                    auto value = std::string(header.value().getStringView());
+                                    // use semicolon as a separator, because pseudo-headers (e.g. :path) have colons (":") in them
+                                    *headers_string += "\t" + key + "; " + value + "\n";
```

Comment:
Should we be concerned about performance here? I'm not sure how string appends work in C++, but in some languages, each append is involves a memory allocation and copy. Suggestion to consider using a string buffer or a join?

Reply:
This might be a safer solution https://stackoverflow.com/a/16029964

I think the only cause for concern here is if `HeaderMap` either has many strings or large ones. If so, then we might want to consider the approach above. If not, then maybe this is just a premature optimisation and we don't need to worry about it.

Reply:
Good callout, but I think this is not super relevant because this transformer is only used during tests -- in this case I think it makes sense to prioritize readability over a handful of milliseconds so that our tests are easier to work with/understand

Reply:
ah, didn't see that this was in tests. fine by me then